LeadKit/LeadKit/Extensions/UIImage/UIImage+Resize.swift

```diff
@@ -28,4 +28,53 @@ public extension UIImage {
         return UIGraphicsGetImageFromCurrentImageContext()
     }
 
+    /**
+     creates a resized copy of an image.
+
+     - parameter size: the new size of the image.
+     - parameter contentMode: the way to handle the content in the new size.
+
+     - returns: a new image
+     */
+    public func resize(size: CGSize,
+                contentMode: UIImageContentMode = .ScaleToFill) -> UIImage? {
+        let horizontalRatio = size.width / size.width
+        let verticalRatio = size.height / size.height
+        var ratio: CGFloat = 1
+
+        switch contentMode {
+        case .ScaleToFill:
+            ratio = 1
+        case .ScaleAspectFill:
+            ratio = max(horizontalRatio, verticalRatio)
+        case .ScaleAspectFit:
+            ratio = min(horizontalRatio, verticalRatio)
+        }
+
+        let rect = CGRect(x: 0, y: 0, width: size.width * ratio, height: size.height * ratio)
+
+        let colorSpace = CGColorSpaceCreateDeviceRGB()
+        let bitmapInfo = CGBitmapInfo(rawValue: CGImageAlphaInfo.PremultipliedLast.rawValue)
+        let context = CGBitmapContextCreate(nil,
```

Comment:
why not
```swift
init?(data: UnsafeMutableRawPointer?, 
width: Int, 
height: Int, 
bitsPerComponent: Int, 
bytesPerRow: Int, 
space: CGColorSpace, 
bitmapInfo: UInt32)
```
?
https://developer.apple.com/reference/coregraphics/cgcontext/1455939-init